protos/sigstore_common.proto

```diff
@@ -169,9 +169,16 @@ message SubjectAlternativeName {
 message X509CertificateChain {
         // The chain of certificates, with indices 0 to n.
         // The first certificate in the array must be the leaf
-        // certificate used for signing. Any intermediate certificates
-        // must be stored as offset 1 to n-1, and the root certificate at
-        // position n.
+        // certificate used for signing.
+        //
+        // Signers MUST NOT include their root CA or CAs in their embedded
+        // certificate chains, and SHOULD NOT include intermediate CAs that
+        // appear in independent roots of trust.
+        //
+        // Verifiers MUST validate the chain carefully to ensure that it
+        // upholds the signer's invariants, and MUST be prepared to handle
```

Comment:
"upholds the signer's invariants" is a little unclear to me. Also, one person's root CA is another person's intermediate CA, so I'd stress "signer" a little bit less.

Maybe:

> Verifiers MUST validate the chain carefully to ensure that it chains up to a root CA that they trust, regardless of whether the chain includes additional intermediate/root CAs. Verifiers MAY enforce additional constraints, such as requiring that all intermediate CAs appear in an independent root of of trust.
>
> Verifiers SHOULD handle old or non-complying bundles that have additional intermediate/root CAs.

Reply:
Sounds good!

Reply:
Done!